doc/examples/example_cmake.sh

```diff
@@ -26,7 +26,7 @@ exec >&2  # redirect output to STDERR
 
 #BEGIN cmake_build
 cd $BLD_DIR
-cmake $SRC_DIR
+cmake -DCMAKE_BUILD_TYPE=PROFILE $SRC_DIR
```

Comment:
What does this do? This snippet is included in the documentation (cookbook section), so any unusual options should be explained there.

Reply:
In the existing example the build is always done with profiling:
https://github.com/gcovr/gcovr/blob/e0b7afef00123b7b6ce4f487a1c4cc9fc60528bc/doc/examples/CMakeLists.txt#L6

To build without profiling you need to change CMakeLists.txt. For this purpose CMake has predefined [build types](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html) which can be extended by user defined ones. Now you can define additional options only for this build type (e.g. `-DCMAKE_BUILD_TYPE=PROFILE`):
https://github.com/gcovr/gcovr/blob/cb8ba6549639463c723246d7e090547ab3c9c1b7/doc/examples/CMakeLists.txt#L7-L8

If you have different toolchains for a cmake you can define the profiling flags in the toolchain file which can be reused in different projects and the project doesn't need to know the flags for each toolchain.

Reply:
aah, OK. I had filtered out any .txt reports for reviewing, but this accidentally prevented me from seeing your change in CMakeLists.txt. Thanks!